internal/pkg/db/redis/data.go

```diff
@@ -1097,13 +1097,25 @@ func deleteEvent(conn redis.Conn, id string) error {
 		return err
 	}
 
+	// Obtain the checksum associated with the event.
+	// This is necessary since the helper functions result in an Event type which does not contain the checksum
```

Comment:
Would it be better to modify the helper functions to return a Redis model instead of a contract?

Reply:
That was my original intent, but that caused other things to break. Rather than digging for a root cause, I went with this direct approach to provide a fix for the bug rather than re-working the whole structure of the Redis DBClient. The problem is that the Redis DBClient is using 3 different representations of the Event type and its not clear which is being used  and we got into this odd situation. Another problem is that the Blackbox tests are not being run, and they seem to not be working so I am very hesitant to start restructuring everything. When I attempted to clean things up the blackbox tests showed even more failures than it is already reporting. I am going to create an issue to go back and re-work the Redis client.

Reply:
Created the issue to clean this up: #2166